Fix lifetimes in SignedCookies/PrivateCookies::get()

akheron opened the pull request:
Before this commit, the `Key` that's passed to `Cookies::signed()` or `Cookies::private()` is forced to have a static lifetime, because the returned `Cookie` gets `self`'s lifetime as its lifetime parameter.

Fix this by using the cookie `name`'s lifetime instead, and just require it to outlive the lifetime of `self`.

This makes it possible e.g. to store `Key` in an axum `Extension` layer instead of a global `Cell`.

imbolc commented:
Would you add a test with a non-static key lifetime? Can't restricting the `name` lifetime cause potential problems? Wouldn't binding the returning cookie to the static lifetime be `fn get(&self, name: &str) -> Option<Cookie<'static>>` be enough?

akheron commented:
Thanks for the review! Changed to use `'static` lifetime, and modified the `remove` tests so that they fail to build without the fix.

I believe that this might also fix issue #15.
